src/utils/locks.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import Package from '../Package';
+import * as logger from './logger';
+import * as npm from './npm';
+import {settleAll} from './promises';
+import {PError} from './errors';
+
+const LOCK_DIST_TAG = 'pyarn-lock';
+
+export async function lock(packages: Array<Package>) {
+  let locks = [];
+  let promises = [];
+
+  logger.info('Attempting to get locks for all packages');
+
+  for (let pkg of packages) {
+    let {name, version} = pkg.config;
+    let promise = npm.info(name).then(pkgInfo => {
```

Comment:
This is the reason we cant *easily* cache the npm.info command, we sometimes want data that could have changed without a version bump (like dist tags).

We could provide two methods though?

Reply:
I think you can ignore caching for now